discovery/gossiper.go

```diff
@@ -570,8 +570,26 @@ func New(cfg Config, selfKeyDesc *keychain.KeyDescriptor) *AuthenticatedGossiper
 		futureMsgs:        newFutureMsgCache(maxFutureMessages),
 		quit:              make(chan struct{}),
 		chanPolicyUpdates: make(chan *chanPolicyUpdateRequest),
-		prematureChannelUpdates: lru.NewCache[uint64, *cachedNetworkMsg]( //nolint: ll
+		prematureChannelUpdates: lru.NewCache(
 			maxPrematureUpdates,
+			lru.WithDeleteCallback(
+				func(k uint64, cmsg *cachedNetworkMsg) {
+					// for every network message which is
+					// not processed, send an error to the
+					// channel.
+					cb := func(msg *networkMsg) {
+						err := fmt.Errorf("premature "+
+							"channel update for "+
+							"peer %x not "+
+							"processed, deleting "+
+							"from LRU cache",
+							msg.peer.PubKey())
+						msg.err <- err
```

Comment:
Can the message be handled already, so `err` channel (of capacity 1) already has an element inside? If the error is not read from the channel on the receiving side (`brontide` package skipped this after `remoteGossipMsgTimeout`), this channel write may block.